src/protobuf-net.Test/Meta/AssemblyMetadataTests.cs

```diff
@@ -0,0 +1,63 @@
+﻿using ProtoBuf.Meta;
+using System;
+using System.Diagnostics;
+using Xunit;
+
+namespace ProtoBuf.unittest.Meta
+{
+    public class AssemblyMetadataTests
+    {
+        [Theory()]
+        [InlineData(null, null, null, null, null, null, "1.2.3.4", "1.2")]
+        [InlineData("", "", "", "", "", "", "1.2.3.4", "1.2")]
+        [InlineData("Company Name", "", "", "Product Name", "Title", "", "1.2.3.4", "1.2")]
+        [InlineData("Company Name", "Copyright", "Description", "Product Name", "Title", "Legal Trademark", "1.2.3.4", "1.2")]
+        public void Test(string companyName, string copyright, string description, string productName, string title,
+            string trademark, string version, string productVersion)
+        {
+            // arrange
+            var outputPath = "AssemblyMetadataTestContract.dll";
+
+            var options = new RuntimeTypeModel.CompilerOptions()
+            {
+                Accessibility = RuntimeTypeModel.Accessibility.Public,
+                TypeName = "Model",
+                OutputPath = outputPath,
+                AssemblyCompanyName = companyName,
+                AssemblyCopyright = copyright,
+                AssemblyDescription = description,
+                AssemblyProductName = productName,
+                AssemblyTitle = title,
+                AssemblyTrademark = trademark,
+                AssemblyVersion = new Version(version),
+                AssemblyProductVersion = new Version(productVersion)
+            };
+            var model = RuntimeTypeModel.Create();
+
+            // act
+            model.Compile(options);
+
+            // assert
+            var fileInfo = FileVersionInfo.GetVersionInfo(outputPath);
+
+            var defaultValue = " ";
+            var expectedVersion        = string.IsNullOrEmpty(version)        ? defaultValue : version;
+            var expectedProductVersion = string.IsNullOrEmpty(productVersion) ? defaultValue : productVersion;
+            var expectedCompanyName    = string.IsNullOrEmpty(companyName)    ? " "          : companyName;
```

Comment:
isn't `" "` the same as `defaultValue`? I wonder whether a local method would help here, though:

```
static string Normalize(string value) => string.IsNullOrEmpty(value) ? " " : value;
...
Assert.Equal(Normalize(copyright), file.LegalCopyright);
// etc
```

Reply:
Good eye, fixed.